pennylane/gradients/general_shift_rules.py

```diff
@@ -378,6 +381,44 @@ def generate_multi_shift_rule(frequencies, shifts=None, orders=None):
     return _combine_shift_rules(rules)
 
 
+def _copy_and_shift_params(tape, indices, shifts, multipliers, cast=False):
+    """Create a copy of a tape and of parameters, and set the new tape to the parameters
+    rescaled and shifted as indicated by ``indices``, ``multipliers`` and ``shifts``."""
+    all_ops = tape.circuit

```

Comment:
Should this be a copy? Without creating a deep copy, any operators that aren't shifted in the new tape will reference the same object as the old tape.

I don't see that being an issue since operators will be immutable, but just curious about opinions.

Reply:
I say no copying. really highlights how much faith we have that we're following our principles :D